sdk/keyvault/ci.yml

```diff
@@ -0,0 +1,39 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
+trigger:
+  branches:
+    include:
+      - master
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - cmake-modules/
+      - eng/
+      - CMakeLists.txt
+      - sdk/core
```

Comment:
Do you really want this pipeline to trigger for core changes as well? If so we might want to actually make the core pipeline build all the libraries instead of having each individual library pipeline trigger.

Reply:
I believe this is something that we need to fix for storage, identity and the template project as well. 
I will create a new issue for that.
(I just mimic what the existing packages are doing for CI)

Reply:
@weshaggard  I was thinking about this one.
I believe we are currently building all service clients (pipelines) for any core change because the Azure Core library is not yet at any package manager where the services can pick a specific version to build with.
This means that for each change on Azure Core, we need to make sure it does not break anything from client services, because if the client service makes a release after a commit to master is done, that release will be using the Azure Core version that comes with it in the Repo.  

At some point, once we publish Azure Core to Vcpkg or someplace, we need to update the services to pull a specific version from Azure Core and release with a dependency to that version. Then on Azure Core, we can forget about running all services for each commit.

Reply:
Why does this show up twice and break threading :(